tests/test_flatten.py

```diff
@@ -31,3 +44,31 @@ def test_flatten_dict():
             "contextInfo_context2": "info2",
         }
     ]
+
+
+def test_flatten_pd():
+    flattened_pd = flatten(
+        PD_DATA, nested_columns=["contextInfo", "neighborhood"]
+    )
+    assert flattened_pd.equals(
+        DataFrame(
+            [
+                {
+                    "answer": 42,
+                    "contextInfo_context1": "info1",
+                    "contextInfo_context2": "info2",
+                }
+            ]
+        )
+    )
+
+
+def test_aux_function_is_called():
```

Comment:
Does the `aux_function` have a name? Do you refer to `is_empty` or `_flatten_df`? 

```suggestion
def test_flatten_df_is_called():
```